File: frontend/src/app/models/post.ts
export type PostType = 'ISSUE' | 'COMPLAINT' | 'SUGGESTION' | 'GENERAL' | 'OTHER';
export type PostStatus = 'DRAFT' | 'SUBMITTED' | 'UNDER_REVIEW' | 'RESOLVED' | 'CLOSED' | 'REJECTED';
export type Priority = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface Post {
  id: number;
  title: string;
  description: string;
  type: PostType;
  status: PostStatus;
  priority?: Priority | null;
  attachmentUrl?: string | null;
  authorId: number;
  authorUsername: string;
  assignedToId?: number | null;
  assignedToUsername?: string | null;
  createdAt: string;
  updatedAt: string;
  resolvedAt?: string | null;
}

File: backend/src/main/java/com/postanything/entity/Post.java
package com.postanything.entity;

import com.postanything.enums.PostStatus;
import com.postanything.enums.PostType;
import com.postanything.enums.Priority;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "posts")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Post {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false, length = 200)
  private String title;

  @Column(nullable = false, length = 5000)
  private String description;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private PostType type;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private PostStatus status;

  @Enumerated(EnumType.STRING)
  private Priority priority;

  @Column
  private String attachmentUrl;

  @ManyToOne(fetch = FetchType.LAZY, optional = false)
  @JoinColumn(name = "author_id")
  private User author;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "assigned_to_id")
  private User assignedTo;

  @Column(nullable = false)
  private LocalDateTime createdAt;

  @Column(nullable = false)
  private LocalDateTime updatedAt;

  private LocalDateTime resolvedAt;
}
